source/Government.cpp

```diff
@@ -562,3 +591,45 @@ bool Government::IsProvokedOnScan() const
 {
 	return provokedOnScan;
 }
+
+
+
+bool Government::ShowReputationTitle(const map<string, map<string, string>> &specialLogs) const
+{
+	if(titleVisibility == TitleVisibility::SHOWN_ON_LOG)
+	{
+		auto it = specialLogs.find("Factions");
+		if(it != specialLogs.end() && it->second.find(GetName()) != it->second.end())
+			return true;
+		return false;
+	}
+	return titleVisibility == TitleVisibility::SHOWN;
+}
+
+
+
+void Government::SetTitleVisibility(bool input)
+{
+	titleVisibility = input ? TitleVisibility::SHOWN : TitleVisibility::HIDDEN;
+}
+
+
+
+string Government::GetReputationTitle() const
+{
+	double rep = Reputation();
+	auto tempTitles = titles;
+	if(tempTitles.empty())
+		tempTitles = GameData::ReputationTitles();
```

Comment:
```suggestion
	auto tempTitles = !titles.empty() ? titles : GameData::ReputationTitles();
```
I think thats the same thing?